### PR TITLE
feat: Queue adapter to choice dynamically

### PR DESCRIPTION
**Pull request by lzianekhodja**

### CODE REVIEW

**Review by alouvion-aneo — DISMISSED** [review]

File: infrastructure/modules/armonik/core-configmap.tf [hunk 1]
```diff
@@ -5,21 +5,21 @@ resource "kubernetes_config_map" "core_config" {
     namespace = var.namespace
   }
   data = merge(var.extra_conf.core, {
-    Components__TableStorage             = local.table_storage_adapter
-    Components__ObjectStorage            = local.object_storage_adapter
-    Components__QueueStorage             = local.queue_storage_adapter
```
Comment (lemaitre-aneo) [inline infrastructure/modules/armonik/core-configmap.tf:10]: I would keep this one to make the infrastructure compatible with both previous and current versions of core.

File: infrastructure/quick-deploy/aws/all/parameters.tfvars [hunk 2]
```diff
@@ -458,19 +458,20 @@ ingress = {
...
-    Amqp__QueueStorage__PollPeriodicity        = "00:00:10"
-    Amqp__QueueStorage__LockRefreshExtension   = "00:02:00"
-    MongoDB__TableStorage__PollingDelayMin     = "00:00:01"
-    MongoDB__TableStorage__PollingDelayMax     = "00:00:10"
-    MongoDB__TableStorage__PollingDelay        = "00:00:01"
-    MongoDB__DataRetention                     = "10.00:00:00"
-    MongoDB__AllowInsecureTls                  = true
-    Redis__Timeout                             = 3000
-    Redis__SslHost                             = ""
+    Components__QueueAdaptorSettings__AdapterAbsolutePath = "/adapters/queue/amqp/ArmoniK.Core.Adapters.Amqp.dll"
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/parameters.tfvars:461]: This should come from the same place than the class name

**Comment by aneojgurhem** [comment]
To me, changes suggested by @lemaitre-aneo are required

File: infrastructure/modules/armonik/core-configmap.tf (hunk 1 above)
Comment (lzianekhodja) [reply infrastructure/modules/armonik/core-configmap.tf:10]: Done

File: infrastructure/quick-deploy/aws/all/parameters.tfvars (hunk 2 above)
Comment (aneojgurhem) [reply infrastructure/quick-deploy/aws/all/parameters.tfvars:461]: Moreover, if the adapter used is rabbitmq, the path should be `/adapters/queue/rabbit/ArmoniK.Core.Adapters.RabbitMQ.dll`

**Review by aneojgurhem — APPROVED** [review]

File: infrastructure/quick-deploy/aws/all/parameters.tfvars (hunk 2 above)
Comment (lzianekhodja) [reply infrastructure/quick-deploy/aws/all/parameters.tfvars]: Done

**Pull request merged**